tests/test_cases/test_cocotb/test_timing_triggers.py

```diff
@@ -194,6 +208,21 @@ async def test_timeout_func_pass(dut):
     assert res == 1
 
 
+@cocotb.test()
+async def test_decorated_coroutine_killed_after_timeout(_):
+    @cocotb.coroutine
+    async def example():
+        await Timer(10, 'ns')
+        return 1
+    coro = example()
+    try:
+        await cocotb.triggers.with_timeout(coro, timeout_time=1, timeout_unit='ns')
+    except cocotb.result.SimTimeoutError:
+        pass
+    assert not coro._finished
+    assert await coro == 1
```

Comment:
Isn't this testing that it wasn't killed?
```suggestion
    assert coro._finished
    assert await coro is None
```

Reply:
```suggestion
    assert coro._finished
    assert await coro == 1
```